Update OpenAPI learn page with IDL support (#5750)
ballerina-platform/ballerina-dev-website: swan-lake/learn-the-platform/ballerina-tooling/openapi-tool.md

>**Info:** For more command options, see [OpenAPI to Ballerina](/learn/cli-documentation/openapi/#openapi-to-ballerina).

### Generate client for IDL import

> With a non-Ballerina Interface Definition Language(IDL) import support, you can add openAPI specification to generate the IDL client and define it as a top-level import to your ballerina file in a below way.

Reviewer: ```suggestion
The non-Ballerina Interface Definition Language(IDL) import feature allows you to add an openAPI specification to generate the IDL client and define it as a top-level import in your Ballerina file as follows.
```

```ballerina
client "./openapi.yaml" as foo;
```

> This IDL client import will be simplified the user experience with respect to the using CLI command.

Reviewer: ```suggestion
This IDL client import will simplify the user experience with respect to using the CLI command.
```

> If you need to add additional details to create your IDL client, use below annotation `@openapi:ClientConfig` with field attributes.

Reviewer: ```suggestion
If you need to add additional details to create your IDL client, use the `@openapi:ClientConfig` annotation below with field attributes.
```

> All of these fields are optional.

Reviewer: ```suggestion
>**Info:** All of these fields are optional.
```

```ballerina
@openapi:ClientConfig {
    tags: ["store"],
    operations: ["op1", "op2"],
    license: "/path/to/license_file.txt",
    isResource: true // (default value => true),
    nullable: false // (default value => false)
}
client "./openapi.yaml" as foo;
```

> >**Info:** For annotation attributes details, see [Annotation reference](#annotation-reference).

Reviewer: ```suggestion
>**Info:** For details of the annotation attributes, see [Annotation reference](#annotation-reference).
```

## Annotation reference

> ### `@openapi:ServiceInfo` annotation

Reviewer: ```suggestion
### The `@openapi:ServiceInfo` annotation
```

The attributes of the annotation are optional and can be used for each particular purpose as described below.  

| Attribute                      | Description                                                                                                                                                                                                                                                                                                                  |
|--------------------------------|------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|

> | `Contract: string?`            | Here, you can provide a path to the OpenAPI contract as a string and the OpenAPI file can either be a `.yaml` or `.json`.                                                                                                                                                                                                    |

Reviewer: ```suggestion
| `Contract: string?`            | Provides the path to the OpenAPI contract as a string and the OpenAPI file can either be a `.yaml` or `.json`.                                                                                                                                                                                                    |
```

> | `Tags: string[]?`              | The compiler only validates resources against operations, which are tagged with a tag specified in the list. If not specified, the compiler validates resources against all the operations defined in the OpenAPI contract.                                                                                                  |

Reviewer: ```suggestion
| `Tags: string[]?`              | Specifies the tag in the list for the compiler to validate resources against operations that are tagged with it. If not specified, the compiler validates resources against all the operations defined in the OpenAPI contract.                                                                                                  |
```

> | `Operations: string[]?`        | This should contain a list of operation names that need to be validated against the resources in the service. If not specified, the compiler validates resources against all the operations defined in the OpenAPI contract.  If both tags and operations are defined, it validates against the union set of the resources.  |

Reviewer: ```suggestion
| `Operations: string[]?`        | Contains a list of operation names that need to be validated against the resources in the service. If not specified, the compiler validates resources against all the operations defined in the OpenAPI contract.  If both tags and operations are defined, it validates against the union set of the resources.  |
```

> | `ExcludeTags: string[]?`       | This stores the tags that do not need to be validated. The annotation can not have both the `excludeTags` and `Tags` attributes at the same time.                                                                                                                                                                            |

Reviewer: ```suggestion
| `ExcludeTags: string[]?`       | Stores the tags that do not need to be validated. The annotation can not have both the `excludeTags` and `Tags` attributes at the same time.                                                                                                                                                                            |
```

> | `ExcludeOperations: string[]?` | This specifies the operations that do not need to be validated.                                                                                                                                                                                                                                                              |

Reviewer: ```suggestion
| `ExcludeOperations: string[]?` | Specifies the operations that do not need to be validated.                                                                                                                                                                                                                                                              |
```

> | `FailOnErrors: boolean?`       | To turn off the validation, add this to the annotation with the value as `false`.                                                                                                                                                                                                                                            |

Reviewer: ```suggestion
| `FailOnErrors: boolean?`       | Turns off the validation when used with `false` in the annotation.                                                                                                                                                                                                                                            |
```

> | `Title: string?`               | Use this to add the title of the `info` section in the generated OpenAPI contract.                                                                                                                                                                                                                                           |

Reviewer: ```suggestion
| `Title: string?`               | Adds the title of the `info` section in the generated OpenAPI contract.                                                                                                                                                                                                                                           |
```

> | `Version: string?`             | Use this to add the version of the `info` section in the generated OpenAPI contract.                                                                                                                                                                                                                                         |

Reviewer: ```suggestion
| `Version: string?`             | Adds the version of the `info` section in the generated OpenAPI contract.                                                                                                                                                                                                                                         |
```

> | `Embed: string?`               | To turn off generating OpenAPI documentation for service for introspection endpoint support, use this attribute with `false` in the annotation.                                                                                                                                                                              |

Reviewer: ```suggestion
| `Embed: string?`               | Turns off generating OpenAPI documentation for the service for introspection endpoint support when used with `false` in the annotation.                                                                                                                                                                              |
```

### `@openapi:ClientConfig` annotation

The `@openapi:ClientConfig` annotation supports for IDL client generation.

```ballerina
@openapi:ClientConfig {
    tags: ["store"],
    operations: ["op1", "op2"],
    license: "/path/to/license_file.txt",
    isResource: true // (default value => true),
    nullable: false // (default value => false)
}
client "./openapi.yaml" as foo;
```

> The attributes of the annotation are optional and can be used for each particular purpose as described below.

Reviewer: ```suggestion
>**Info:** The attributes of the annotation are optional and can be used for each particular purpose as described below.
```

| Attribute                      | Description                                                                                                                                                                                                                    |
|--------------------------------|--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|

> | `Tags: string[]?`              | Generate only remote/resources functions for operations, which are tagged with a tag specified in the list. If not specified, the generator generates remote/resources for all the operations defined in the OpenAPI contract. |

Reviewer: ```suggestion
| `Tags: string[]?`              | Generate only the remote/resources functions for operations that are tagged with a tag specified in the list. If not specified, the generator generates remote/resources for all the operations defined in the OpenAPI contract. |
```

> | `Operations: string[]?`        | This should contain a list of operation names that need to be included in the client. If not specified, the generator generates remote/resources for all the operations defined in the OpenAPI contract.                       |

Reviewer: ```suggestion
| `Operations: string[]?`        | Contains a list of operation names that need to be included in the client. If not specified, the generator generates remote/resources for all the operations defined in the OpenAPI contract.                       |
```

> | `License: string?`             | Here, you can provide a path to the license as a string and the license file should be a text file.                                                                                                                            |

Reviewer: ```suggestion
| `License: string?`             | Provides a path to the license as a string and the license file should be a text file.                                                                                                                            |
```

> | `isResource: boolean?`         | To select client methods as remote functions for generation, add this to the annotation with the value `false`.                                                                                                                |

Reviewer: ```suggestion
| `isResource: boolean?`         | Specifies the client methods selected as remote functions for generation when used with `false` in the annotation..                                                                                                                |
```

> | `Nullable: boolean?`           | To enable generating all data types in the record with Ballerina nil support, add this to the annotation with the value `true`.                                                                                                |  

Reviewer: ```suggestion
| `Nullable: boolean?`           | Enables generating all data types in the record with Ballerina nil support, when used with `true` in the annotation..                                                                                                |  
```
